torch/distributed/_fsdp/fully_sharded_data_parallel.py

```diff
@@ -628,6 +631,71 @@ def forward(self, *args: Any, **kwargs: Any) -> Any:
 
         return outputs
 
+    @torch.no_grad()
+    def _write_back_current_shard(self):
+        for p in self.params:
+            if not p._is_sharded:
+                pass
+            chunks = p._full_param_padded.chunk(self.world_size)
+            if len(chunks) < (self.rank + 1):
+                # if there are not enough chunks, there's nothing to writeback here
```

Comment:
in which cases can this happen?

Reply:
If there are more workers than elements in the tensor we run into this situation.

Reply:
possibly we should not support this use case? data is already not what users expected, right?

Reply:
@zhaojuanmao this behavior is supported by FSDP and is currently used by the test suite. test_fsdp_overlap.py module `Layer` has a one element parameter that will trigger this behavior. I C&P that code from `_get_shard` but have not added a test covering it.

Reply:
when Layer only has one element, I thought _full_param_padded has already been padded for every rank. So this case should not be triggered at all here...?

Reply:
You're right, `_full_param_padded` is already padded so that condition will never happen.